File: python/신고 결과 받기.py
# def solution(id_list, report, k):
#     answer = [0] * len(id_list)
#     reports = {x:0 for x in id_list}

#     for y in set(report):
#         reports[y.split()[1]] += 1
        
#     for z in set(report):
#         if reports[z.split()[1]] >= k:
#             answer[id_list.index(z.split()[0])] += 1
#     return answer

def solution(id_list, report, k):
    report_dict = {user: set() for user in id_list}
    report_count = {user: 0 for user in id_list}
    
    for elem in report:
        reporter, reported = elem.split(' ')
        if reported not in report_dict[reporter]:
            report_dict[reporter].add(reported)
            report_count[reported] += 1
    
    banned_user = {user for user, count in report_count.items() if count >= k}
    answer = [0] * len(id_list)
    for i, user in enumerate(id_list):
        answer[i] += len(report_dict[user] & banned_user)
    return answer

File: js/신고 결과 받기.js
function solution(id_list, report, k) {

    const reportDict = new Map();
    const reportCount = new Map();

    
    
    id_list.forEach((user) => {
        reportDict.set(user, new Set());
        reportCount.set(user ,0);
    });
    
    report.forEach((elem) => {
        const [reporter, reported] = elem.split(' ');
        if (!reportDict.get(reporter).has(reported)) {
            reportDict.get(reporter).add(reported);
            reportCount.set(reported, reportCount.get(reported) + 1);
        }
    })
    
    const bannedUsers = new Set(
        Array.from(reportCount.entries())
            .filter(([user, count]) => count >= k)
            .map(([user]) => user)
    );
    
    return id_list.map((user) => {
        return Array.from(reportDict.get(user)).filter((reported) => {
            return bannedUsers.has(reported)
        }).length;
    });
}